dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/FeatureUploadConfiguration.kt

```diff
@@ -9,10 +9,8 @@ package com.datadog.android.v2.api
 /**
  * Contains the upload configuration for an [FeatureScope] instance.
  *
- * @property endpointUrl the url endpoint data should be uploaded to
- * @property payloadFormat the expected format of the payload
+ * @property requestFactory creates a request from a given batch and its metadata
  */
 data class FeatureUploadConfiguration(
-    val endpointUrl: String,
-    val payloadFormat: PayloadFormat
+    val requestFactory: RequestFactory
```

Comment:
Yeah, even though it will be empty for now (I don't foresee any instant need) I think it makes sense as a convention. It coherently separates "storage" and "upload" configurations which is one of the goals in V2. While the `requestFactory` answers "what" to upload, we will need `FeatureUploadConfiguration` as soon as we run into "how" (e.g. how frequent or what timeout, like @0xnm said).

I think we will keep it on iOS as it adds good verbosity in code and tests (e.g. mocking `.noOp()` upload configuration will make it clear that the uploader is muted).